packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2818,33 +2834,21 @@ export class ContainerRuntime
 				this.deltaManager.minimumSequenceNumber;
 		}
 
-		// Surround the actual processing of the operation with messages to the schedule manager indicating
```

Comment:
There is a bunch of duplicated logic between observeNonRuntimeMessage() and processRuntimeMessage().
Maybe created beforeProcessingMessage() & afterProcessingMessage() and move most of the common logic there (to be called from both places)? More chances that if we need to make changes, they will apply to both paths.

Reply:
Yes, there is. We are in the middle of refactoring this code I believe. I don't want to change it in this PR. I will consider doing a follow up to clean this up more. Mark also has some comments about refactoring and I can do a refactoring PR later.

Reply:
@pragya91 split them apart but left them (nearly) identical for the express purpose of having them be separate so we could carefully remove functionality from the `observe` function.  So let's not combine them unless we have changed our mind about diverging treatment of these two groups.